src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -0,0 +1,714 @@
+
+@doc raw"""
+    CompressibleEulerMulticomponentEquations2D
+
+  !!! warning "Experimental code"
+    This system of equations is experimental and can change any time.
+
+Multicomponent version of the compressible Euler equations for an ideal gas in two space dimensions.
+"""
+struct CompressibleEulerMulticomponentEquations2D{RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, 5}
+  gamma1          ::RealT 
+  gamma2          ::RealT
+  gas_constant1   ::RealT
+  gas_constant2   ::RealT
+  cv1             ::RealT
+  cv2             ::RealT
+  cp1             ::RealT
+  cp2             ::RealT
+end
+
+function CompressibleEulerMulticomponentEquations2D()
+
+  # Set ratio of specific heat of the gas per species 
+  # (It holds: gamma = cp/cv) (Can be used for consistency check!)
+  gamma1  = 1.4
+  gamma2  = 1.4
+  
+  # Set specific gas constant with respect to the molar mass per species 
+  # (It holds: gas_constant = R/M, with R = molar gas constant, M = molar mass) 
+  # (For a calorically and thermally perfect gas Mayer's relation holds: gas_constant = cp - cv) (Can be used for consistency check!)
+  gas_constant1     = 0.4
+  gas_constant2     = 0.4
+
+  # Set specific heat for a constant volume per species
+  cv1    = 1.0
+  cv2    = 1.0
+
+  # Set specific heat for a constant pressure per species
+  cp1    = 1.4
+  cp2    = 1.4
+
+  CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+end
+
+
+function CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2)
+  
+  # Set specific heat for a constant volume per species
+  cv1    = gas_constant1 / (gamma1 - 1.0)
+  cv2    = gas_constant2 / (gamma2 - 1.0)
+
+  # Set specific heat for a constant pressure per species
+  cp1    = gas_constant1 + cv1 
+  cp2    = gas_constant2 + cv2
+
+  CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+end
+
+
+function CompressibleEulerMulticomponentEquations2D(check, gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+  # check variable is just a dummy to prevent recursion
+
+  if isapprox(gamma1, cp1/cv1; atol = 0.01) 
```

Comment:
```suggestion
  if !isapprox(gamma1, cp1/cv1) 
```
If I were you, I'd not specify a tolerance explicitly unless you know why and what you are doing. Also, you need to add the negation operator `!` before `isapprox` (also in the checks below), otherwise the comparisons don't make sense...

Reply:
fixed in 9424975